front/pages/join.js

```diff
@@ -12,6 +28,48 @@ export default (formType) => {
         zipCodeField = dom('#become_adherent_address_postalCode');
     }
 
+    const autocompeleteWrapper = dom('.address-autocomplete');
+    const addressBlock = dom('.address-block');
+    const fieldCityName = dom('#adherent_registration_address_cityName');
+    const fieldRegion = dom('#adherent_registration_address_region');
+
+    const address = new AddressObject(
+        dom('#adherent_registration_address_address'),
+        dom('#adherent_registration_address_postalCode'),
+        fieldCityName,
+        dom('#adherent_registration_address_country'),
+        dom('#adherent_registration_address_region')
+    );
+
+    if (!address.isFilled()) {
+        if ('undefined' === typeof google) {
+            return;
+        }
+        const autocomplete = new GooglePlaceAutocomplete(autocompeleteWrapper, address, 'form form--full form__field');
```

Comment:
I think we can refactor this part by moving this code into a new class (for ex: `AutocompletedAddressForm`) and avoid the code duplication